diffrax/integrate.py

```diff
@@ -210,9 +521,19 @@ def _apply_history(x):
         keep = lambda a, b: jnp.where(keep_step, a, b)
         y = jtu.tree_map(keep, y, state.y)
         solver_state = jtu.tree_map(keep, solver_state, state.solver_state)
-        made_jump = keep(made_jump, state.made_jump)
         solver_result = keep(solver_result, RESULTS.successful)
 
+        # updating discontinuity
+        if delays is not None:
+            found_jump = lambda a, b: jnp.where(discont_update, a, b)
+            discontinuities = found_jump(
+                state.discontinuities.at[state.discontinuities_save_index].set(tnext),
+                state.discontinuities,
+            )
+            discontinuities_save_index = found_jump(
+                state.discontinuities_save_index + 1, state.discontinuities_save_index
+            )
```

Comment:
Note for technical reasons to do with how the stepping works, we need to take a bit of care around making in-place updates. This is the helper function you need:
https://github.com/patrick-kidger/diffrax/blob/b8475527eacdba81328130edd7dadd08a0b34063/diffrax/integrate.py#L324

This is to work around a limitation of the XLA compiler. I've got a PR here:
https://github.com/google/jax/pull/13062
that should eventually fix this, and mean we can tidy things up here. But we're not there yet.

Reply:
alright i'll create another helper function since the conditioning is not the same as `maybe_inplace`
```py 
def maybe_inplace_delay(i, x, u):
            return inplace(x).at[i].set(jnp.where(discont_update, u, x[i]))
```
and make the appropriate changes